Sources/JWTKeychain/Support/Configuration/Configuration.swift

```diff
@@ -15,67 +15,103 @@ public protocol ConfigurationType {
-    
+
     /// Key used to sign the JWT
     public var signatureKey: String
-    
+
     /// Key used to check the signing the JWT
     public var publicKey: String? = nil
-    
+
     /// Which signer will be used while signing the JWT
     public var signer: String
 
-    private init(signer: String, signatureKey: String, publicKey: String?, secondsToExpire: Double){
+    public enum Error: Swift.Error {
+        case noJWTConfig
+        case missingConfig(String)
+    }
+
+    public init(drop: Droplet) throws {
+        try self.init(config: drop.config)
+    }
+
+    public init(config: Config) throws {
+
+        guard let jwtConfig = config["jwt"]?.object else {
+            throw Error.noJWTConfig
+        }
+
+        guard let signer: String = jwtConfig["signer"]?.string else {
+            throw Error.missingConfig("signer")
+        }
+
+        guard let secondsToExpire = jwtConfig["secondsToExpire"]?.double else {
+            throw Error.missingConfig("secondsToExpire")
+        }
+
+        guard let signatureKey = jwtConfig["signatureKey"]?.string else {
+            throw Error.missingConfig("signatureKey")
+        }
+
+        let publicKey: String? = jwtConfig["publicKey"]?.string
+
+        if publicKey == nil {
+            //The ECDSA and RSA (ES*/RS*) signers take a private key for signing and needs a matching public key for verifying.
+            if signer.hasPrefix("ES") || signer.hasPrefix("RS"){
+                throw Error.missingConfig("publicKey")
+            }
+        }
+
+        self.init(signer: signer, signatureKey: signatureKey, publicKey: publicKey, secondsToExpire: secondsToExpire)
+
+    }
+
+    public init(signer: String, signatureKey: String, publicKey: String?, secondsToExpire: Double){
         self.signer = signer
         self.signatureKey = signatureKey
         self.publicKey = publicKey
-        self.secondsToExpire = secondsToExpire}
-    // Register configs
-    public static func boot(signer: String, signatureKey: String, publicKey: String?, secondsToExpire: Double) throws {
-        Configuration.instance = Configuration(signer: signer, signatureKey: signatureKey, publicKey: publicKey, secondsToExpire: secondsToExpire)
-    }
-    
-    
+        self.secondsToExpire = secondsToExpire
+      }
+
     /// Gets token signature key
     ///
     /// - Returns: signature key
-    public static func getTokenSignatureKey() -> Bytes {
-        
-        return Array(Configuration.instance!.signatureKey.utf8)
-        
+    public func getTokenSignatureKey() -> Bytes {
+        return Array(self.signatureKey.utf8)
+
     }
-    
+
     /// Gets token public key
     ///
     /// - Returns: public key
     /// - Throws: if cannot retrieve signature key
-    public static func getTokenPublicKey() throws -> Bytes {
-        return Array(Configuration.instance!.publicKey!.utf8)
-        
+    public func getTokenPublicKey() throws -> Bytes {
+        return Array(self.publicKey!.utf8)
```

Comment:
What should I do to avoid that? 
right now there is a nil check before using the method.
```
if self.publicKey != nil {

                key = try self.getTokenPublicKey()
            }
```